### ISSUE TITLE
On demand cleaning requires 2 clicks

### ISSUE THREAD

**Issue by Joel889** [Lusito/forget-me-not#115]
This is regarding local storage / IndexDB:
These aren't cleaned automatically (unless the browser is restarted). But they can be cleaned manually using the addon. If I click the on demand cleaning button, and then look at Firefox's storage management interface, local storage / IndexDB entries are still present. I have to click it a second time (or twice in a row), in order for local storage / IndexDB to disappear from Firefox's storage management.

All manual clean up options are selected (except passwords + protect open domains (meaning they will be deleted)).

**Comment by Lusito** [comment]
Anything below the storage folder is indexedDB as far as I can tell and has nothing to do with localstorage.

Since this is a really simple call to a simple API, I think this must be a Firefox bug. I haven't been able to reproduce this so far. What do you mean by 5-10 entries? individual websites? (i.e. folders in the storage/default folder)

**Comment by Joel889** [comment]
Yes, wait until there are several folders under \storage\default. While the existence of several folders by itself is probably not what is causing, it increases the likelihood of the bug occurring. Not sure if it only happens for some sites and the way they are setting indexDB or some other combination.

Also, the following addon is clearing indexDB reliably; both on demand and also on startup:
https://addons.mozilla.org/en-US/firefox/addon/storagerazor/
